lib/Runtime/Debug/TTSnapObjects.cpp

```diff
@@ -2117,15 +2128,277 @@ namespace TTD
+                NSSnapValues::EmitTTDVar(sgi->arguments_values[i], writer, i != 0 ? NSTokens::Separator::CommaSeparator : NSTokens::Separator::NoSeparator);
+            }
+            writer->WriteSequenceEnd();
+
+            writer->WriteUInt32(NSTokens::Key::u32Val, sgi->byteCodeReader_offset, NSTokens::Separator::CommaSeparator);
+        }
+
+        void ParseAddtlInfo_SnapGeneratorInfo(SnapObject* snpObject, FileReader* reader, SlabAllocator& alloc) {
+            SnapGeneratorInfo* sgi = alloc.SlabAllocateStruct<SnapGeneratorInfo>();
+
+            sgi->generatorPrototype = reader->ReadInt32(NSTokens::Key::i32Val, true);
+
+            sgi->frame_slotCount = reader->ReadInt32(NSTokens::Key::i32Val, true);
+
+            if (sgi->frame_slotCount == 0) {
+                sgi->frame_slotArray = nullptr;
+            }
+            else {
+                sgi->frame_slotArray = alloc.SlabAllocateArray<TTDVar>(sgi->frame_slotCount);
+            }
+            reader->ReadKey(NSTokens::Key::frame_slotArray, true);
+            reader->ReadSequenceStart();
+            for (Js::RegSlot i = 0; i < sgi->frame_slotCount; ++i) {
+                sgi->frame_slotArray[i] = NSSnapValues::ParseTTDVar(i != 0, reader);
+            }
+            reader->ReadSequenceEnd();
+
+
+            sgi->state = reader->ReadUInt32(NSTokens::Key::u32Val, true);
+            sgi->scriptFunction = reader->ReadAddr(NSTokens::Key::objectId, true);
+
+            uint32 callInfoLength = reader->ReadLengthValue(true);
+            reader->ReadSequenceStart_WDefaultKey(true);
+            TTDAssert(callInfoLength == sizeof(Js::CallInfo), "unexpected mismatch in length between sizeof(CallInfo) and value serialized to log");
+            for (uint32 i = 0; i < callInfoLength; i++) {
+                sgi->arguments_callInfo[i] = reader->ReadNakedByte(i != 0);
+            }
+            reader->ReadSequenceEnd();
+
+            sgi->arguments_count = reader->ReadUInt32(NSTokens::Key::u32Val, true);
+
+            if (sgi->arguments_count == 0) {
+                sgi->arguments_values = nullptr;
+            }
+            else {
+                sgi->arguments_values = alloc.SlabAllocateArray<TTDVar>(sgi->arguments_count);
+            }
+            reader->ReadKey(NSTokens::Key::arguments_values, true);
+            reader->ReadSequenceStart();
+            for (uint32 i = 0; i < sgi->arguments_count; ++i) {
+                sgi->arguments_values[i] = NSSnapValues::ParseTTDVar(i != 0, reader);
+            }
+            reader->ReadSequenceEnd();
+
+            sgi->byteCodeReader_offset = reader->ReadUInt32(NSTokens::Key::u32Val, true);
+
+            SnapObjectSetAddtlInfoAs<SnapGeneratorInfo*, SnapObjectType::SnapGenerator>(snpObject, sgi);
+        }
+
+#if ENABLE_SNAPSHOT_COMPARE
+        void AssertSnapEquiv_SnapGeneratorInfo(const SnapObject *sobj1, const SnapObject *sobj2, TTDCompareMap &compareMap) {
```

Comment:
Looks like this needs a "todo" to be implemented, just making sure that mappings between the objects are consistent?

Reply:
fixed